### ISSUE TITLE
Crash/Coredump/SIGSEGV/StackTrace From PVR While Drive Is Under Heavy Load

### ISSUE THREAD

**Issue by vindicatorr** [xbmc/xbmc#16937]
### Describe the bug
While viewing a recorded video from tvheadend (also on the same system), I did an action not related to Kodi which made heavy use of the mechanical hard drive.
Immediately, a coredump for kodi was initiated.

## Expected Behavior

A SIGSEGV should not occur when buffering, but rather just wait and maybe have a waiting signal/symbol/animation occur.
> CRenderManager::WaitForBuffer - timeout waiting for buffer

## Actual Behavior

SIGSEGV, causing a coredump and subsequent termination of Kodi.

## Possible Fix

You'll know better than I. Good luck!

### To Reproduce

1. Play a video on Kodi where the video also resides on the same system.
2. Do some action (cp, dd) that takes up a lot of hard drive I/O.
3. Profit.

### Debuglog

The debuglog can be found here:
[kodi_crashlog-20191117_170549.log](https://github.com/xbmc/xbmc/files/3856254/kodi_crashlog-20191117_170549.log)

### Your Environment
Used Operating system:

 - [ ] Android
 - [ ] iOS
 - [X] Linux
 - [ ] OSX
 - [ ] Raspberry-Pi
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: ArchLinux
 - Kodi version: 19.0-ALPHA1 Git:20191031-cce8119892-dirty

**Comment by vindicatorr** [comment]
Ok, so it may not be a resource/buffer issue...
For this crash, I had finished watching a recorded video, deleted it, started watching the next video, put it on pause while I focused on reading some stuff online.
When I was ready to continue watching the show, I saw kodi had disappeared... then I saw another crash had happened.

I don't recall having done any heavy I/O action (nor does journalctl show anything screwy), and you can see the timing between me pausing and a stack trace occurring was about 1 hour. It is now about 25 minutes later when I came to the realization and posting this.
[kodi_crashlog-20191118_190032.log](https://github.com/xbmc/xbmc/files/3861469/kodi_crashlog-20191118_190032.log)

**Comment by vindicatorr** [comment]
Yup, 5.3.2 installed and it also crashed just now.
[kodi_crashlog-20191120_000902.log](https://github.com/xbmc/xbmc/files/3867597/kodi_crashlog-20191120_000902.log)
Odd that the stack trace notes 5.2.0 still, even when it says 5.3.2 later as well as in kodi.
I went ahead and renamed that file to *.bak now.

**Comment by DaVukovic** [comment]
Coudl you please do the following things: 

1. enable debuglogging like: 

```
<advancedsettings>
  <loglevel>1</loglevel>
</advancedsettings>

By doing it in that way, you won't see the overlay at the screen but the logfile will be more informative.

2. try to use an older ffmpeg version. The version you are using might not be supported. After buildig Kodi on Ubuntu I have: 

FFmpeg version/source: 4.0.4-Kodi

**Comment by ksooo** [comment]
All three logs show crashes at completely different places. It will be really hard to make any sense out of this. Just saying.

**Comment by vindicatorr** [comment]
Sure. I just had a crash while I was away (kodi was left open, but not playing anything I don't believe. A machine was delivered and I had to run an errand).
[kodi_crashlog-20191120_102142.tar.gz](https://github.com/xbmc/xbmc/files/3870529/kodi_crashlog-20191120_102142.tar.gz)
I set "Enable debug logging" for this output. It was 340KB, so I zipped it. (Appreciated the mention of the possibility to not have the overlay, but it's not a big deal for a test).

I'll rebuild kodi to use an older ffmpeg (assuming that can be done) later. Right now I have a tree I need to take down since I'm paying for that lift.

**Comment by vindicatorr** [comment]
Actually DaVukovic, do you really think it's about the ffmpeg version?

I had cloned the repository and was about to build your version based off of the PKGBUILD, but got to thinking that it's really only been recently that Kodi has been crashing as much as it has (6 times since the 15th). There were a few more times at the beginning of November, the time before that was one at the end of September, 2 in the middle and none in the 30 days prior.

I looked at my pacman cache and see "ffmpeg-1:4.2-4-x86_64.pkg.tar.xz" was installed at the beginning of August (again, no crashes in coredumpctl), then "ffmpeg-1:4.2.1-1-x86_64.pkg.tar.xz" on September 7th, which would align with those 2 crashes on the 11th and 12th.
I wouldn't think these tiny revision upgrades (latest "ffmpeg-1:4.2.1-4-x86_64.pkg.tar.xz") would have something so gnarly in them to break Kodi.

I mean it's one thing to rename a function or something for a major or maybe minor version change. But for a tiny revision, I'd think it'd only change the instructions within a function and have ffmpeg fail, but not Kodi (if Kodi is catching exceptions from peripheral libraries).

Basically, I just don't want to be running a fool's errand if it is just that (a guess). I'm just not savvy enough to interpret the cause of the crash in the bt, so if you can show me why you think it may be ffmpeg, I'm all ears.

You say it's ffmpeg, ksooo says it's crashing at different places, but I'm seeing it say it's Thread 1, pointing to libfreetype. So I just don't know what to think here.

**Comment by DaVukovic** [comment]
Oh, it's for sure only a guess.

**Comment by vindicatorr** [comment]
It doesn't seem to be a problem now. Closing.
